src/core/router/identity.cc

```diff
@@ -730,27 +730,23 @@ Keys CreateRandomKeys() {
 IdentHash CreateRoutingKey(
     const IdentHash& ident) {
   std::uint8_t buf[41];  // ident + yyyymmdd
-  memcpy(buf, (const std::uint8_t *)ident, 32);
+  std::memcpy(buf, (const std::uint8_t *)ident, 32);
   time_t t = time(nullptr);
   struct tm tm;
-  // TODO(unassigned): never use sprintf, use snprintf instead.
+  std::stringstream ss;
+  ss.imbue(std::locale(std::locale::classic()));
 #ifdef _WIN32
-  gmtime_s(&tm, &t);
-  sprintf_s(
-      reinterpret_cast<char *>((buf + 32)),
-      9,
-      "%04i%02i%02i",
-      tm.tm_year + 1900,
-      tm.tm_mon + 1,
-      tm.tm_mday);
+  gmtime_s(&t, &tm);
+  tm.tm_year += 1900;
+  tm.tm_mon += 1;
+  ss << std::put_time(&tm, "%Y%m%d");
+  std::memcpy(reinterpret_cast<char *>((buf + 32)), ss.str().data(), 9);
```

Comment:
If you're going to keep the memcpy's then you should setup the empty buffer with `ByteStream` and refactor accordingly.

As for keeping a stringstream for time, that's TBD depending on what else you plan on doing.

Reply:
> If you're going to keep the memcpy's then you should setup the empty buffer with ByteStream and refactor accordingly.

Will do, I'll look at setting this up with `ByteStream`. Will try both refactors (memcpy and Boost.DateTime), and see which comes out cleaner.

Thanks for the review, and the pointers on where to go next :+1: